client/gribiclient.go

```diff
@@ -799,7 +813,7 @@ func (c *Client) clearPendingOp(op *spb.AFTResult) (*OpResult, error) {
 	case spb.AFTResult_FIB_PROGRAMMED, spb.AFTResult_FIB_FAILED:
 		delete(c.qs.pendq.Ops, op.Id)
 	case spb.AFTResult_RIB_PROGRAMMED:
-		if c.state.SessParams.GetAckType() != spb.SessionParameters_RIB_AND_FIB_ACK {
+		if c.state.SessParams.GetAckType() != spb.SessionParameters_RIB_AND_FIB_ACK || c.state.SessParams.GetAckType() == spb.SessionParameters_RIB_AND_FIB_ACK && TreatRIBACKAsCompletedInFIBACKMode {
```

Comment:
With `TreatRIBACKAsCompletedInFIBACKMode == true` we delete the op from the pending q. What happens if the FIB ACK comes in later? Do we error out? (sorry I'm too lazy to check the code in the night)

Reply:
Good question -- and this turned out to be a bug, thanks for the thought.

It also made me go and implement sending both `RIB_PROGRAMMED` /and/ `FIB_PROGRAMMED` to the reference implementation. PTAL.

Reply:
I added a test case that covers this.